pontoon/terminology/management/commands/import_microsoft_terminology.py

```diff
@@ -0,0 +1,185 @@
+
+
+
+
+class Command(BaseCommand):
+    """
+    Import terminology files for a set of locales or all of them in the concurrent fashion.
+    """
+    help = 'Import all or selected terminology files from the Microsoft Language Portal.'
+
+
+    def add_arguments(self, parser):
+        # A number of processes that will be spawned to download and parse terms from tbx files.
+        parser.add_argument('--workers', type=int, default=10, help='A number of download workers.')
+
+        # A list of locales that user want to import. Fetches all files if the user doesn't pass any parameter.
+        parser.add_argument('locales', nargs='*', help='A list of locales to download.')
+
+
+    def handle(self, *args, **options):
+        locale_codes = options.get('locales')
+        workers_num = options.get('workers')
+        Term.objects.term_index = CachedDBTermIndex
+
+        # A list of Pontoon locales that share terminology with Microsoft Portal.
+        terminology_locales = (
+            Locale
+                .objects
+                .filter(ms_terminology_code__isnull=False)
+                .exclude(ms_terminology_code='')
+                .order_by('code')
+        )
+
+        if locale_codes:
+            locales_to_download = terminology_locales.filter(code__in=locale_codes)
+            available_locale_codes = set(terminology_locales.values_list('code', flat=True))
+            unmatched_locales = set(locale_codes) - available_locale_codes
+
+            if unmatched_locales:
+                log.error("Can't find following locales: {}".format(
+                    ','.join(unmatched_locales)
+                ))
+                log.info('Available locales:')
+                log.info(', '.join(sorted(available_locale_codes)))
+                sys.exit()
+
+        else:
+            # Download all locales if no specific locale is requested.
+            locales_to_download = terminology_locales
+
+        pool = ProcessPoolExecutor(workers_num)
+        terms = {}
+        try:
+            terms_per_locale = pool.map(
+                retrieve_microsoft_terms,
+                [
+                    (locale.code, locale.ms_terminology_code)
+                    for locale in locales_to_download
+                ]
+            )
+        except ValueError:
+            # During testing I noticed that MS Terminology Portal returns 50x errors
+            # or just ''.
+            log.error("Couldn't load data from server. Please, try again later.")
+            sys.exit(-1)
+
+        # Merge all terms imported from different locale files
+        for locale_terms in terms_per_locale:
+            for term in locale_terms:
+                if term.term_id in terms:
+                    terms[term.term_id].translations.update(
+                       term.translations
+                    )
+                else:
+                    terms[term.term_id] = term
+
+        # filter terms with MS specific phrases
```

Comment:
Filter.

Move belo the `log.info` statements, right above `cleaned_terms` definition.